src/main/ruby/core/regexp.rb

```diff
@@ -381,3 +406,36 @@ def to_s
     self[0]
   end
 end
+
+Truffle::KernelOperations.define_hooked_variable(
+  :$~,
+  -> b { Truffle::RegexpOperations.last_match(b) },
+  -> v, b { Truffle::RegexpOperations.set_last_match(v, b) })
+
+Truffle::KernelOperations.define_hooked_variable(
+  :$`,
```

Comment:
Could you try quoting this and the `$'`? That would likely fix the bad highlighting at least on GitHub.